conans/test/functional/toolchains/cmake/test_cmake_toolchain_win_clang.py

```diff
@@ -50,35 +57,203 @@ def build(self):
+    @pytest.mark.tool_clang(version="12")  # repeated, for priority over the mingw64 clang
+    def test_clang_cmake_runtime_version(self, client):
+        generator = "Ninja"
+        # Make sure that normal CMakeLists with verify=False works
+        client.save({"CMakeLists.txt": gen_cmakelists(verify=False, appname="my_app",
+                                                      appsources=["src/main.cpp"])})
+        client.run("create . pkg/0.1@ -pr=clang -s compiler.runtime=dynamic -s compiler.cppstd=17 "
+                   "-s compiler.runtime_version=v142 "
+                   '-c tools.cmake.cmaketoolchain:generator="{}"'.format(generator))
+
+        assert 'cmake -G "{}"'.format(generator) in client.out
+        assert "GNU-like command-line" in client.out
+        assert "main __clang_major__12" in client.out
+        # Check this! Clang compiler in Windows is reporting MSC_VER and MSVC_LANG!
+        assert "main _MSC_VER192" in client.out
+        assert "main _MSVC_LANG2017" in client.out
+        assert "main _M_X64 defined" in client.out
+        assert "main __x86_64__ defined" in client.out
+        cmd = re.search(r"MYCMD=(.*)!", str(client.out)).group(1)
+        cmd = cmd + ".exe"
+        check_vs_runtime(cmd, client, "16", build_type="Release", static_runtime=False)
+
+
 @pytest.mark.skipif(platform.system() != "Windows", reason="requires Win")
-def test_clang(client):
-    client.run("create . pkg/0.1@ -pr=clang")
-    # clang compilations in Windows will use MinGW Makefiles by default
-    assert 'cmake -G "MinGW Makefiles"' in client.out
-    assert "main __clang_major__12" in client.out
-    # Check this! Clang compiler in Windows is reporting MSC_VER and MSVC_LANG!
-    assert "main _MSC_VER19" in client.out
-    assert "main _MSVC_LANG2014" in client.out
+class TestVSClangCL:
+    """
+    This is also LLVM/Clang, but distributed with the VS installation
+    """
+    @pytest.mark.tool_cmake(version="3.23")
+    @pytest.mark.tool_visual_studio(version="17")
+    def test_clang_visual_studio_generator(self, client):
+        """ This is using the embedded ClangCL compiler, not the external one"""
+        generator = "Visual Studio 17"
+        client.run("create . pkg/0.1@ -pr=clang -s compiler.runtime=dynamic "
+                   "-s compiler.cppstd=17 -s compiler.runtime_version=v143 "
+                   '-c tools.cmake.cmaketoolchain:generator="{}"'.format(generator))
+        assert 'cmake -G "{}"'.format(generator) in client.out
+        assert "MSVC-like command-line" in client.out
+        assert "main __clang_major__14" in client.out
+        # Check this! Clang compiler in Windows is reporting MSC_VER and MSVC_LANG!
+        assert "main _MSC_VER193" in client.out
+        assert "main _MSVC_LANG2017" in client.out
+        assert "main _M_X64 defined" in client.out
+        assert "main __x86_64__ defined" in client.out
+        assert "-m64" not in client.out
```

Comment:
If this test were to run on arm64 Windows, I think `-m64` is not passed anymore

Reply:
This is ok, it is VS generator, should not pass ``-m64``